File: Cpp/125_Valid_Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int i=0;
        int j=s.size()-1;
        while(i<j){
            while(!isAlphanumericChar(s[i]) && i<j) ++i;
            while(!isAlphanumericChar(s[j]) && i<j) --j;
            
            if(tolower(s[i]) != tolower(s[j])){
                return false;
            }
            ++i;
            --j;
        }
        
        return true;
    }
    
    bool isAlphanumericChar(char ch){
        return (ch >= 'a' && ch <= 'z')
            || (ch >= 'A' && ch <= 'Z')
            || (ch >= '0' && ch <= '9');
    }
};

File: Kotlin/125_Valid_Palindrome.kt
class Solution {
    fun isPalindrome(s: String): Boolean {
        var i = 0
        var j = s.length - 1
        while(i < j){
            while (i < j && !s[i].isLetterOrDigit()) ++i
            while (i < j && !s[j].isLetterOrDigit()) --j
            
            if(s[i].toLowerCase() != s[j].toLowerCase()) return false
            ++i
            --j
        }
        
        return true
    }
}